app/models/post.rb

```diff
@@ -1,2 +1,18 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
+
+  after_create :update_post_counter
+  after_destroy :update_post_counter
+
```

Comment:
- [OPTIONAL] It would be great to add the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call #update_post_counter method every time a new post has been saved to the database. To do this you should add a single line of code, public, like this:
`after_save :update_post_counter`